src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/QuickInfoSourceProvider.QuickInfoSource.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.QuickInfo;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.Text.Shared.Extensions;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.Text;
+using Roslyn.Utilities;
+
+using IntellisenseQuickInfoItem = Microsoft.VisualStudio.Language.Intellisense.QuickInfoItem;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.QuickInfo
+{
+    internal partial class QuickInfoSourceProvider
+    {
+        private class QuickInfoSource : IAsyncQuickInfoSource
+        {
+            private readonly QuickInfoSourceProvider _quickInfoSourceProvider;
+            private readonly ITextBuffer _subjectBuffer;            
+            private IDocumentProvider _documentProvider = new DocumentProvider();
```

Comment:
You don't seem to be using this, I'm guessing it's dead. (More generally looks like this existed to support unit testing...once we fix signature help it might be killable entirely.)

Reply:
deleted